esp/logging/logginglib/modularlogagent.cpp

```diff
@@ -0,0 +1,1526 @@
+            result = false;
+        }
+        if (checkTime &&
+            !(modifiers.count('H') && modifiers.count('M') && modifiers.count('S')))
+        {
+            uerrlog(Major, "'%s' value '%s' missing required time modifiers", xpath, format.str());
+            result = false;
+        }
+    }
+    return result;
+}
+
+/**
+ * Sets up properties used by the target to manage files, including:
+ *   - concurrent files: the total number of open files the agent will maintain at one time
+ *   - rollover interval: the frequency at which one file will be closed and another opened
+ *     in its place
+ *   - rollover size: the maximum size, in bytes, a log file should reach before being closed
+ *     and replaced by another; the first write into an empty file is unconstrained
+ */
+bool CFileTarget::configureFileHandling(const IPTree& configuration)
+{
+    bool result = true;
+
+    StringBuffer concurrentFiles(configuration.queryProp(xpathConcurrentFiles));
+    if (concurrentFiles.isEmpty())
+        concurrentFiles.append(defaultConcurrentFiles);
+    if (TokenDeserializer()(concurrentFiles, m_concurrentFiles) != Deserialization_SUCCESS)
+    {
+        uerrlog(Major, "'%s' is an invalid value for '%s'", concurrentFiles.str(), xpathConcurrentFiles);
+        result = false;
+    }
+    else if (0 == m_concurrentFiles)
+    {
+        uerrlog(Major, "'%s' cannot be zero", xpathConcurrentFiles);
+        result = false;
+    }
+
+    const char* propValue = configuration.queryProp(xpathRolloverInterval);
+    if (isEmptyString(propValue))
+        propValue = defaultRolloverInterval;
+    if (streq(propValue, rolloverInterval_Daily))
+        m_rolloverInterval = DailyRollover;
+    else if (streq(propValue, rolloverInterval_None))
+        m_rolloverInterval = NoRollover;
+    else
+    {
+        uerrlog(Major, "'%s' is an invalid value for '%s'", propValue, xpathRolloverInterval);
+        m_rolloverInterval = UnknownRollover;
+        result = false;
+    }
+
+    propValue = configuration.queryProp(xpathRolloverSize);
+    if (isEmptyString(propValue))
+        propValue = defaultRolloverSize;
+    if (!extractSize(propValue, m_rolloverSize))
+    {
+        uerrlog(Major, "'%s' is an invalid value for '%s'", propValue, xpathRolloverSize);
+        result = false;
+    }
+
+    return result;
+}
+
+/**
+ * Sets up the file path pattern to be used by the target.
+ */
+bool CFileTarget::configurePattern(const IPTree& configuration)
+{
+    bool result = true;
+
+    const char* propValue = configuration.queryProp(xpathFilePathPattern);
+    if (isEmptyString(propValue))
+    {
+        uerrlog(Major, "'%s' is a required configuration value", xpathFilePathPattern);
+        result = false;
+    }
+    else
+    {
+        m_pattern.setown(new Pattern(*this));
+        if (!m_pattern->setPattern(propValue))
+            result = false;
+    }
+
+    return result;
```

Comment:
I do personally like the return statements to be inline.  It makes it more intuitive to me that there is no fall through behavior.  I can quickly see where the function exits rather than seeing if anything else happens even thought the return value has been set.

It's not a big deal.  

But I think it is common in the HPCC code if it's not too complex to return when you are done.  Again there are cases where it does have to fall through or returning early just doesn't work well.

Reply:
Also, consistency doesn't have to come before intuitiveness or readability.  I mean if you return inline from a simple function it doesn't mean you always have to return inline.  It's meant to be something that either helps keep the code clear or it doesn't.